### PR TITLE
Upgrade to Kustomize v5.0.0

### PR DESCRIPTION
**Pull request by starcraft66**
Kustomize 5.0.0 contains important fixes for loading remote resources and other stuff.

### CODE REVIEW

**Review by devstein — COMMENTED** [review]
@starcraft66 Thanks for making a PR! I love Starcraft 😁 

Excited that there is a new kustomize release. Can you also update [	`sigs.k8s.io/kustomize/api` to v0.13.0](https://github.com/kubernetes-sigs/kustomize/releases/tag/api%2Fv0.13.0) in the go.mod file and run `go mod tidy`?

We may also need to drop support for Go 1.18: https://github.com/kubernetes-sigs/kustomize/pull/4877

**Comment by starcraft66** [comment]
> I love Starcraft 😁
That's awesome, glad to hear it! I do too obviously

> Excited that there is a new kustomize release. Can you also update [ `sigs.k8s.io/kustomize/api` to v0.13.0](https://github.com/kubernetes-sigs/kustomize/releases/tag/api%2Fv0.13.0) in the go.mod file and run `go mod tidy`?
> 
> We may also need to drop support for Go 1.18: [kubernetes-sigs/kustomize#4877](https://github.com/kubernetes-sigs/kustomize/pull/4877)

I went ahead and did all of that, also fixed a couple unit tests that broke.

File: test/krm/behaviors/generate-resources.yaml [hunk 1]
```diff
@@ -10,6 +10,4 @@ metadata:
           # otherwise, path should be relative to manifest files, like
           # path: ../../../ksops
 files:
-  # NOTE: File is relative to the /base manifest!
```
Comment (starcraft66) [inline test/krm/behaviors/generate-resources.yaml:13]: I believe this is actually a bug that was fixed in https://github.com/kubernetes-sigs/kustomize/pull/4654

Comment (devstein) [reply test/krm/behaviors/generate-resources.yaml:13]: Nice! You're right

**Review by devstein — APPROVED** [review]
Awesome work. Thank you for contributing!

**Pull request merged**